### PR TITLE
adds text replace option incl signin-widget version lookup

### PR DESCRIPTION
**Pull request by swiftone**
## Description:
- **What's changed?** 

    - Adds the ability to replace `-=OKTA_REPLACE_WITH_XXX=-` variables in text, regardless of any markdown formatting/code blocks. 
    - Adds a lookup for the latest version of a given signin-widget major version.  This will let us automatically update references to the latest whenever any D.O.C. update happens.  (Note: Changes in major versions likely change the instructions so those remain manual)
    - Adds unit tests for this sort of internal tooling

- **Is this PR related to a Monolith release?** 

    - No

### Resolves:

* [OKTA-270398](https://oktainc.atlassian.net/browse/OKTA-270398)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/test.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Hello world!
+
+-=OKTA_REPLACE_WITH_WIDGET_VERSION=-
+
+-=OKTA_REPLACE_WITH_WIDGET_VERSION=- "-=OKTA_REPLACE_WITH_WIDGET_VERSION=-"
+-=OKTA_REPLACE_WITH_TEST_JUNK=-
+```
+-=OKTA_REPLACE_WITH_WIDGET_VERSION=-
+```
```
Comment (bretterer) [inline packages/@okta/vuepress-site/docs/reference/test.md]: We should only have these at `/packages/@okta.vuepress-site/test_page/index.md`

File: packages/@okta/vuepress-site/.vuepress/scripts/build-replacement-vars.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+const thingsToReplace = { 
+  /* KEYS HERE GET WRAPPED IN '-=OKTA_REPLACE_WITH_XXX=-'
+   *
+   * Changes WILL require restarting `yarn dev` :(
+   */
+
+/* VALUES BELOW THIS LINE INTENDED FOR EASY EDITING */
+  WIDGET_VERSION: 3, // Leave as major version
+  TEST_JUNK: 'this is a test replacement', // Leave for testing
```
Comment (bretterer) [inline packages/@okta/vuepress-site/.vuepress/scripts/build-replacement-vars.js]: Can these be pulled in via `/.vuepress/config.js`?

File: packages/@okta/vuepress-site/docs/reference/test.md (hunk 1 above)
Comment (swiftone) [reply packages/@okta/vuepress-site/docs/reference/test.md]: D'oh!  This was my early test file.  I even checked _twice_ to make sure I hadn't included it.  Sigh.

File: packages/@okta/vuepress-site/.vuepress/scripts/build-replacement-vars.js (hunk 2 above)
Comment (swiftone) [reply packages/@okta/vuepress-site/.vuepress/scripts/build-replacement-vars.js]: Can do.

**Pull request merged**